src/modifier/modifier-type.ts

```diff
@@ -121,18 +121,41 @@ export class ModifierType {
    * Populates item tier for ModifierType instance
    * Tier is a necessary field for items that appear in player shop (determines the Pokeball visual they use)
    * To find the tier, this function performs a reverse lookup of the item type in modifier pools
+   * It checks the weight of the item and will use the first tier for which the weight is greater than 0
+   * This is to allow items to be in multiple item pools depending on the conditions, for example for events
+   * If all tiers have a weight of 0 for the item, the first tier where the item was found is used
    * @param poolType Default 'ModifierPoolType.PLAYER'. Which pool to lookup item tier from
+   * @param party optional. Needed to check the weight of modifiers with conditional weight (see {@linkcode WeightedModifierTypeWeightFunc})
+   *  if not provided or empty, the weight check will be ignored
+   * @param rerollCount Default `0`. Used to check the weight of modifiers with conditional weight (see {@linkcode WeightedModifierTypeWeightFunc})
    */
-  withTierFromPool(poolType: ModifierPoolType = ModifierPoolType.PLAYER): ModifierType {
+  withTierFromPool(poolType: ModifierPoolType = ModifierPoolType.PLAYER, party?: PlayerPokemon[], rerollCount: number = 0): ModifierType {
+    let defaultTier: undefined | ModifierTier;
     for (const tier of Object.values(getModifierPoolForType(poolType))) {
       for (const modifier of tier) {
         if (this.id === modifier.modifierType.id) {
-          this.tier = modifier.modifierType.tier;
-          return this;
+          let weight: number;
+          if (modifier.weight instanceof Function) {
+            weight = party ? (modifier.weight as Function)(party, rerollCount) : 0;
+          } else {
+            weight = modifier.weight as number;
```

Comment:
```suggestion
            weight = modifier.weight;
```
Same here, TS already knows it's a number because of the `if` statement.